### ISSUE TITLE
Feature Request: Support forFeatureAsync to handle Hooks with injectables

### ISSUE THREAD

**Issue by nartc** [kpfromer/nestjs-typegoose#394]
Same as `nestjs/mongoose` https://docs.nestjs.com/techniques/mongodb#hooks-middleware

This will allow for something like the following:

```ts
TypegooseModule.forFeatureAsync([
   {
      // some properties here,
      inject: [authConfiguration.KEY],
      useFactory: (authConfig: AuthConfig) => {
         const schema = someMethodToGetSchema();
         schema.pre('save', function(next) {
            if (this.isModified('password')) {
               this.password = someHashFunction(authConfig.salt)
               next()
            }
         })
         return schema;
      }
   }
])
```

**Comment by smolinari** [comment]
Question: isn't the facility in Typegoose for using hooks not enough?

Scott

**Comment by nartc** [comment]
@smolinari is the question directed to me? if yes, then no, hooks alone aren’t enough. This feature request allows for using Dependency Injection configurations in the hooks which isn’t possible atm.

**Comment by smolinari** [comment]
Ah, Gotcha. I would have put that password hashing logic in a service anyway. But, I guess that's just me. 😁 

Scott

**Comment by nartc** [comment]
@smolinari Right. I used to do the same but then the logic gets repetitive (while I also tried to DRY it). 

There are some different use-cases like post-processing some field changed. Eg: a wysiwyg editor sending raw markdown to the backend, the backend post-processes the markdown and saves html in the db. Whenever this field gets updated, run the post-processing again. Stuffs like this make a lot of sense in a format of hooks and being able to inject services when constructing hooks would be a lot nicer, and the logic is encapsulated.

**Comment by 87789771** [comment]
@nartc Hello, how did you solve it?

**Comment by smolinari** [comment]
@87789771 - You do the hashing in either the User or Auth services, where it actually belongs. 

Scott
